packages/neuron-ui/src/utils/hooks.ts

```diff
@@ -393,3 +394,46 @@ export const useOnHandleNetwork = ({ history }: { history: ReturnType<typeof use
     },
     [history]
   )
+
+export const useChainTypeByGenesisBlockHash = (url: string | null, cb: (chainType: ChainType) => void) => {
+  const timerRef = useRef<NodeJS.Timeout | undefined>()
+  const BUFFER_TIME = 200
+
+  useEffect(() => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current)
+    }
+
+    if (url) {
+      timerRef.current = setTimeout(() => {
```

Comment:
Could you please clarify a bit on why we need `setTimeout` here? Is it because it needs to wait for the connection to the CKB node?
I am concerned the use of `setTimeout` would introduce some stability issues at a certain stage. Are there other ways to achieve the same purpose without using the `setTimeout`?

Reply:
> Could you please clarify a bit on why we need `setTimeout` here? Is it because it needs to wait for the connection to the CKB node?
> I am concerned the use of `setTimeout` would introduce some stability issues at a certain stage. Are there other ways to achieve the same purpose without using the `setTimeout`?

It works as a `debounce` method to reduce unexpected HTTP requests.